subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/BuildOperationReportingCompiler.java

```diff
@@ -0,0 +1,127 @@
+            @Override
+            public BuildOperationDescriptor.Builder description() {
+                return BuildOperationDescriptor.displayName("Invoke compiler for " + task.getIdentityPath())
+                    .details(new Details());
+            }
+
+            @Override
+            public WorkResult call(BuildOperationContext context) {
+                WorkResult result = delegate.execute(spec);
+                context.setResult(toBuildOperationResult(result));
+                return result;
+            }
+
+            private Result toBuildOperationResult(WorkResult result) {
+                if (result instanceof CompilationWithAnnotationProcessingResult) {
+                    AnnotationProcessingResult annotationProcessingResult = ((CompilationWithAnnotationProcessingResult) result).getAnnotationProcessingResult();
+                    List<AnnotationProcessorDetails> details = new ArrayList<AnnotationProcessorDetails>();
+                    for (AnnotationProcessorResult processorResult : annotationProcessingResult.getAnnotationProcessorResults()) {
+                        details.add(toAnnotationProcessorDetails(processorResult));
+                    }
+                    return new Result(details);
+                }
+                return new Result(null);
+            }
+
+            private DefaultAnnotationProcessorDetails toAnnotationProcessorDetails(AnnotationProcessorResult result) {
+                return new DefaultAnnotationProcessorDetails(result.getClassName(), result.isIncremental(), result.getExecutionTimeInMillis());
+            }
+        });
+    }
+
+    private static class Details implements CompileWithAnnotationProcessingBuildOperationType.Details {
+    }
+
+    private static class Result implements CompileWithAnnotationProcessingBuildOperationType.Result {
+
+        private final List<AnnotationProcessorDetails> annotationProcessorDetails;
+
+        Result(List<AnnotationProcessorDetails> annotationProcessorDetails) {
+            this.annotationProcessorDetails = annotationProcessorDetails;
+        }
+
+        @Override
+        public List<AnnotationProcessorDetails> getAnnotationProcessorDetails() {
+            return annotationProcessorDetails;
+        }
+
+    }
+
+    private static class DefaultAnnotationProcessorDetails implements AnnotationProcessorDetails {
+
+        private final String className;
+        private final boolean incremental;
+        private final long executionTimeInMillis;
+
+        DefaultAnnotationProcessorDetails(String className, boolean incremental, long executionTimeInMillis) {
+            this.className = className;
+            this.incremental = incremental;
+            this.executionTimeInMillis = executionTimeInMillis;
+        }
+
+        @Override
+        public String getClassName() {
+            return className;
+        }
+
+        @Override
+        public boolean isIncremental() {
```

Comment:
Sorry I had just mentioned this as an example of why we may want a dedicated object, not to include it right away. If we wanted to add this, we would expose the actual kind, i.e. isolating or aggregating.

Reply:
Well, I figured it would actually be helpful to drive adoption of incremental annotation processing so I decided to add it right away. Would you report the declared type (aggregating, isolating, dynamic, unknown) or the "resolved" type, i.e. aggregating/isolating/unknown instead of dynamic?

Reply:
The resolved one.

Reply:
So, I'll create a new enum for the build operation with three values `AGGREGATING`, `ISOLATING`, `UNKNOWN` (or `NON_INCREMENTAL`?) and a `getType()` method in `AnnotationProcessorDetails`.

Reply:
Done in 4fbd219538b.